js/woo.js

```diff
@@ -0,0 +1,1465 @@
+        //     success: true
+        // }
+        const transactions = this.safeValue (response, 'rows', {});
+        return this.parseTransactions (transactions, currency, since, limit);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        // example in fetchTransactions
+        const networkizedCode = this.safeString (transaction, 'token');
+        if (currency === undefined) {
+            currency = this.getCurrencyByNetworkizedCode (networkizedCode);
+        }
+        const code = currency['code'];
+        let movementDirection = this.safeStringLower (transaction, 'token_side');
+        if (movementDirection === 'withdraw') {
+            movementDirection = 'withdrawal';
+        }
+        let fee_token = this.safeString (transaction, 'fee_token', '');
+        if (fee_token === '') {
+            fee_token = code;
+        }
+        const addressTo = this.safeString (transaction, 'target_address');
+        const addressFrom = this.safeString (transaction, 'source_address');
+        const timestamp = this.safeTimestamp (transaction, 'created_time');
+        return {
+            'id': this.safeString (transaction, 'id'),
+            'txid': this.safeString (transaction, 'tx_id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': movementDirection === 'deposit' ? addressTo : addressFrom, // TODO - should become deprecated property
+            'address_from': addressTo, // TODO - make unified property
+            'address_to': addressFrom, // TODO - make unified property
+            'tag': this.safeString (transaction, 'extra'),
+            'type': movementDirection,
+            'amount': this.safeNumber (transaction, 'amount'),
+            'currency': code,
+            'status': this.parseTransactionStatus (this.safeString (transaction, 'status')),
+            'updated': this.safeTimestamp (transaction, 'updated_time'),
+            'fee': {
+                'currency': fee_token,
+                'cost': this.safeNumber (transaction, 'fee_amount'),
+                'rate': undefined,
+            },
+            'info': transaction,
+        };
+    }
+
+    parseTransactionStatus (status) {
+        const statuses = {
+            'NEW': 'pending',
+            'CONFIRMING': 'pending',
+            'PROCESSING': 'pending',
+            'COMPLETED': 'ok',
+            'CANCELED': 'canceled',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async fetchTransfers (code = undefined, since = undefined, limit = undefined, params = {}) {
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+        }
+        const request = {
+            'type': 'COLLATERAL',
+        };
+        params = this.omit (params, 'type'); // conflict! we need to change 'type' param for spot/futures into 'exchangeType'
```

Comment:
There is no conflict here, cause fetchTransfers does not operate with markets, it operates with currencies which are usually working exchange-wide. Let's cut out lines 1222-1226.